### PR TITLE
feat(pandas): support memtable in pandas backend

### LINKED ISSUE #5467: feat(pandas): support `memtable` in pandas backend
### What happened?

While working on #5466 I discovered that `ibis.memtable` does not work with the pandas or dask backends:

```python
>>> import ibis
>>> expr = ibis.memtable({"column": [0, 1]})
>>> ibis.get_backend(expr)
<ibis.backends.duckdb.Backend object at 0x1288d7e90>
>>> expr.execute()
   column
0       0
1       1
>>> ibis.set_backend("pandas")
>>> expr.execute()
Traceback (most recent call last):
  Cell In[6], line 2
    expr.execute()
  File ~/code/ibis/ibis/expr/types/core.py:300 in execute
    return self._find_backend(use_default=True).execute(
  File ~/code/ibis/ibis/backends/pandas/__init__.py:240 in execute
    return execute_and_reset(node, params=params, **kwargs)
  File ~/code/ibis/ibis/backends/pandas/core.py:489 in execute_and_reset
    result = execute(
  File ~/mambaforge/envs/ibisdev/lib/python3.11/site-packages/multipledispatch/dispatcher.py:278 in __call__
    return func(*args, **kwargs)
  File ~/code/ibis/ibis/backends/pandas/trace.py:136 in traced_func
    return func(*args, **kwargs)
  File ~/code/ibis/ibis/backends/pandas/core.py:435 in main_execute
    return execute_with_scope(
  File ~/code/ibis/ibis/backends/pandas/core.py:222 in execute_with_scope
    result = execute_until_in_scope(
  File ~/code/ibis/ibis/backends/pandas/trace.py:136 in traced_func
    return func(*args, **kwargs)
  File ~/code/ibis/ibis/backends/pandas/core.py:354 in execute_until_in_scope
    result = execute_node(
  File ~/mambaforge/envs/ibisdev/lib/python3.11/site-packages/multipledispatch/dispatcher.py:278 in __call__
    return func(*args, **kwargs)
  File ~/code/ibis/ibis/backends/pandas/trace.py:136 in traced_func
    return func(*args, **kwargs)
  File ~/code/ibis/ibis/backends/pandas/dispatch.py:32 in raise_unknown_table_node
    raise com.UnboundExpressionError(
UnboundExpressionError: Node of type 'PandasInMemoryTable' has no data bound to it. You probably tried to execute an expression without a data source.
```

dask has a similar problem, probably because it delegates the operation to pandas.

I have also tried with sqlite and polars which work as expected. For backends that do not support the operation we get a more explicit exception, for instance for datafusion:

<details>

```python
Traceback (most recent call last):
  Cell In[13], line 1
    expr.execute()
  File ~/code/ibis/ibis/expr/types/core.py:300 in execute
    return self._find_backend(use_default=True).execute(
  File ~/code/ibis/ibis/backends/datafusion/__init__.py:254 in execute
    output = self.to_pyarrow(expr, params=params, limit=limit, **kwargs)
  File ~/code/ibis/ibis/backends/base/__init__.py:266 in to_pyarrow
    self.to_pyarrow_batches(expr, params=params, limit=limit, **kwargs)
  File ~/code/ibis/ibis/backends/datafusion/__init__.py:244 in to_pyarrow_batches
    frame = self._get_frame(expr, params, limit, **kwargs)
  File ~/code/ibis/ibis/backends/datafusion/__init__.py:213 in _get_frame
    return self.compile(expr, params, **kwargs)
  File ~/code/ibis/ibis/backends/datafusion/__init__.py:272 in compile
    return translate(expr.op())
  File ~/mambaforge/envs/ibisdev/lib/python3.11/functools.py:909 in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
  File ~/code/ibis/ibis/backends/datafusion/compiler.py:24 in operation
    raise com.OperationNotDefinedError(f'No translation rule for {type(op)}')
OperationNotDefinedError: No translation rule for <class 'ibis.backends.pandas.client.PandasInMemoryTable'>
```

</details>

### What version of ibis are you using?

ibis master

### What backend(s) are you using, if any?

pandas
dask

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by anjakefala**
Closes #5467

(Relatively new contributor, and welcome feedback on anything nuanced that I am missing here!)

### CODE REVIEW

**Comment by anjakefala** [comment]
I see that I have some tests to remove `notimpl` from; I had only checked in `test_generic.py`.

It seems I should also remove them for the `dask` plugin.

**Comment by cpcloud** [comment]
Thanks for the PR!

Except for the `XPASS`ing tests, which are hopefully pretty easy to fix (just remove the corresponding backend from the list!) this looks good to me!

**Comment by cpcloud** [comment]
I tried out 

```
pytest 'ibis/backends/tests/test_client.py::test_interactive_repr_max_columns[pandas-False]'
```

And that passes for me:

**Comment by cpcloud** [comment]
commitlint is failing because

```
tests(pandas): memtable for pandas backend is now supported
```

should be

```
test(pandas): ...
```

File: ibis/backends/tests/test_array.py [hunk 1]
```diff
@@ -473,11 +473,8 @@ def test_array_slice(con, start, stop):
 )
 @pytest.mark.notimpl(
     ["dask", "pandas"],
-    raises=com.UnboundExpressionError,
-    reason=(
-        "Node of type 'InMemoryTable' has no data bound to it. "
-        "You probably tried to execute an expression without a data source."
-    ),
+    raises=com.OperationNotDefinedError,
+    reason=("Operation 'ArrayMap' is not implemented for this backend"),
```
Comment (cpcloud) [inline ibis/backends/tests/test_array.py:477]:
```suggestion
    reason="Operation 'ArrayMap' is not implemented for this backend",
```

File: ibis/backends/tests/test_array.py [hunk 2]
```diff
@@ -508,11 +505,8 @@ def test_array_map(backend, con):
 )
 @pytest.mark.notimpl(
     ["dask", "pandas"],
-    raises=com.UnboundExpressionError,
-    reason=(
-        "Node of type 'InMemoryTable' has no data bound to it. "
-        "You probably tried to execute an expression without a data source."
-    ),
+    raises=com.OperationNotDefinedError,
+    reason=("Operation 'ArrayMap' is not implemented for this backend'"),
```
Comment (cpcloud) [inline ibis/backends/tests/test_array.py:509]:
```suggestion
    reason="Operation 'ArrayMap' is not implemented for this backend",
```

**Review by cpcloud — APPROVED** [review]
Couple of micronits, not blocking.

Merging on green!

**Comment by cpcloud** [comment]
No need to squash!

**Review by cpcloud — APPROVED** [review]

**Comment by anjakefala** [comment]
I am also running into a `DeprecationWarning` with one of our dependencies and `pytest` locally, that I worked around with `pytest -sv -m pandas ibis/backends/tests/ -W ignore::DeprecationWarning`

Example:

```
ibis/backends/tests/test_temporal.py:34: in <module>
    from google.api_core.exceptions import BadRequest as GoogleBadRequest
../../miniconda3/envs/ibis-dev/lib/python3.10/site-packages/google/api_core/exceptions.py:29: in <module>
    from google.rpc import error_details_pb2
../../miniconda3/envs/ibis-dev/lib/python3.10/site-packages/google/rpc/__init__.py:18: in <module>
    import pkg_resources
../../miniconda3/envs/ibis-dev/lib/python3.10/site-packages/pkg_resources/__init__.py:3324: in <module>
    def _initialize_master_working_set():
../../miniconda3/envs/ibis-dev/lib/python3.10/site-packages/pkg_resources/__init__.py:3298: in _call_aside
    f(*args, **kwargs)
../../miniconda3/envs/ibis-dev/lib/python3.10/site-packages/pkg_resources/__init__.py:3349: in _initialize_master_working_set
    tuple(dist.activate(replace=False) for dist in working_set)
../../miniconda3/envs/ibis-dev/lib/python3.10/site-packages/pkg_resources/__init__.py:3349: in <genexpr>
    tuple(dist.activate(replace=False) for dist in working_set)
../../miniconda3/envs/ibis-dev/lib/python3.10/site-packages/pkg_resources/__init__.py:2870: in activate
    declare_namespace(pkg)
../../miniconda3/envs/ibis-dev/lib/python3.10/site-packages/pkg_resources/__init__.py:2338: in declare_namespace
    warnings.warn(msg, DeprecationWarning, stacklevel=2)
E   DeprecationWarning: Deprecated call to `pkg_resources.declare_namespace('sphinxcontrib')`
```

**Comment by cpcloud** [comment]
That's something we can address in a separate PR, we usually do that by including a specific pattern after the `"ignore"` string in `pyproject.toml` so that we aren't propagating a bunch of useless warning spam to users from other libraries.

**Comment by cpcloud** [comment]
It doesn't seem to be triggering the failure in CI or locally for me, so it could be an environment issue.

**Comment by cpcloud** [comment]
🚢-ing it!

**Pull request merged**